src/main/java/com/google/devtools/build/lib/remote/RemoteSpawnRunner.java

```diff
@@ -413,18 +415,31 @@ static Action buildAction(
     return action.build();
   }
 
-  static Platform parsePlatform(Label platformLabel, @Nullable String platformDescription) {
+  static Platform parsePlatform(@Nullable PlatformInfo executionPlatform, @Nullable String defaultPlatformProperties) {
     Platform.Builder platformBuilder = Platform.newBuilder();
-    try {
-      if (platformDescription != null) {
-        TextFormat.getParser().merge(platformDescription, platformBuilder);
+
+    if (executionPlatform != null && !Strings.isNullOrEmpty(executionPlatform.remoteExecutionProperties())) {
+      // Try and get the platform info from the execution properties.
+      try {
+        TextFormat.getParser().merge(executionPlatform.remoteExecutionProperties(), platformBuilder);
+      } catch (ParseException e) {
+        throw new IllegalArgumentException(
```

Comment:
I think it would be better to not use a runtime exception, as this will kill the Bazel server. Prefer an `AbruptExitException` or similar to provide a user friendly error.

Reply:
Change to UserExecError, since the various remote classes already throw and handle ExecError.